### PR TITLE
Casts a repo object before checking permission

### PR DESCRIPTION
**Pull request by git-hyagi**
This change is similar to #4933, but for the condition function where domain is not enabled.
[noissue]

fixes: #5619

### CODE REVIEW

**Comment by mdellweg** [comment]
This sounds reasonable, but also looks like a bugfix to me.
In that case we absolutely need a changelog.

**Comment by lubosmj** [comment]
@git-hyagi, to which pulpcore versions we should backport this to make it available for pulp-container?

File: CHANGES/5619.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Fixed a bug where containers couldn't be built from repository versions when using RBAC.
```
Comment (lubosmj) [inline CHANGES/5619.bugfix:1]: Please, make this changelog more generic. Many people have no idea what is the container building in this context. Is it about building pulp-oci-images?

**Comment by git-hyagi** [comment]
> @git-hyagi, to which pulpcore versions we should backport this to make it available for pulp-container?

I think to any version >=3.49  (3.49 and 3.55, right?) which is the lower bound of pulp-container plugin: https://github.com/pulp/pulp_container/blob/main/requirements.txt#L2C1-L2C17

File: CHANGES/5619.bugfix [hunk 2]
```diff
@@ -0,0 +1 @@
+Fixed a bug in a permission function to make it work with plugin Repositories.
```
Comment (mdellweg) [inline CHANGES/5619.bugfix:1]: Can you name the actual function? It is user visible inside of the access policy, right?

**Review by mdellweg — APPROVED** [review]

**Review by lubosmj — APPROVED** [review]
Thank you!

**Pull request merged**